osquery/tables/system/windows/battery.cpp

```diff
@@ -0,0 +1,264 @@
+      LOG(ERROR) << "Failed to get battery device detail: " << GetLastError();
+      continue;
+    }
+
+    // Enumerated a battery.  Ask it for information.
+    HANDLE hBattery = CreateFile(pdidd->DevicePath,
+                                 GENERIC_READ | GENERIC_WRITE,
+                                 FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                 nullptr,
+                                 OPEN_EXISTING,
+                                 FILE_ATTRIBUTE_NORMAL,
+                                 nullptr);
+    if (hBattery == INVALID_HANDLE_VALUE) {
+      LOG(ERROR) << "Failed to open handle for battery device "
+                 << wstringToString(pdidd->DevicePath) << ": "
+                 << GetLastError();
+      continue;
+    }
+    auto const hBatteryGuard =
+        scope_guard::create([&]() { CloseHandle(hBattery); });
+
+    // Ask the battery for its tag - needed for later queries
+    BATTERY_QUERY_INFORMATION bqi = {0};
+    DWORD dwWait = 0; // do not wait for a battery, return immediately
+    DWORD dwOut;
+    if (!(DeviceIoControl(hBattery,
+                          IOCTL_BATTERY_QUERY_TAG,
+                          &dwWait,
+                          sizeof(dwWait),
+                          &bqi.BatteryTag,
+                          sizeof(bqi.BatteryTag),
+                          &dwOut,
+                          nullptr) &&
+          bqi.BatteryTag)) {
+      LOG(ERROR) << "Failed to get tag for battery device "
+                 << wstringToString(pdidd->DevicePath) << ": "
+                 << GetLastError();
+      continue;
+    }
+
+    BATTERY_INFORMATION bi = {0};
+    bqi.InformationLevel = BatteryInformation;
+    if (DeviceIoControl(hBattery,
+                        IOCTL_BATTERY_QUERY_INFORMATION,
+                        &bqi,
+                        sizeof(bqi),
+                        &bi,
+                        sizeof(bi),
+                        &dwOut,
+                        nullptr)) {
+      // Only non-UPS system batteries count
+      if (!(bi.Capabilities & BATTERY_SYSTEM_BATTERY) ||
+          (bi.Capabilities & BATTERY_IS_SHORT_TERM)) {
+        continue;
+      }
+
+      if (bi.Capabilities & BATTERY_CAPACITY_RELATIVE) {
+        LOG(WARNING) << "Battery is reporting in unknown (relative) units. "
+                        "Values may not be in mAh, mA, and mV.";
+      }
+
+      Row row;
+
+      // Some possible values for chemistry, though we already have
+      // seen LiP which is not listed
+      // https://learn.microsoft.com/en-us/windows/win32/power/battery-information-str
+      row["chemistry"] = SQL_TEXT(bi.Chemistry);
+
+      // Assume that 12 volts is the intended voltage for the
+      // battery in order to convert from the mWh units that
+      // Microsoft provides to match the mAh units that the battery
+      // table already uses for macOS.
+      const int designedVoltage = 12;

```

Comment:
How confident are you here? And why?

Reply:
I was not able to find any API that provides the designed voltage, so I'm not sure what to do here to make the values on Windows match the values on macOS besides trying this assumption. I also only have a single Windows device for testing, so my inclination is to see how these values do in the real world with bigger Windows deployments.